source/GLaDOS.cpp

```diff
@@ -81,77 +76,78 @@ bool GMController::init(std::vector<std::shared_ptr<DonutModel>> d,
  *
  * This method is used to run the GM for generating and managing current ship events
  */
-void GMController::update(float dt) {
+void GLaDOS::update(float dt) {
 	// Removing breaches that have 0 health left
 	for (int i = 0; i < MAX_EVENTS; i++) {
-		if (breaches.at(i) == nullptr) {
+		if (ship->getBreaches().at(i) == nullptr) {
 			continue;
 		}
-		if (breaches.at(i)->getHealth() == 0) {
-			breaches.at(i)->setAngle(-1);
+		if (ship->getBreaches().at(i)->getHealth() == 0) {
+			ship->getBreaches().at(i)->setAngle(-1);
 			breachFree.at(i) = true;
 		}
 	}
 
 	// Remove doors that have been resolved and opened. Also raise doors that are resolved.
 	for (int i = 0; i < MAX_DOORS; i++) {
-		if (doors.at(i) == nullptr) {
+		if (ship->getDoors().at(i) == nullptr) {
 			continue;
 		}
-		if (doors.at(i)->resolvedAndRaised()) {
-			doors.at(i)->setAngle(-1);
+		if (ship->getDoors().at(i)->resolvedAndRaised()) {
+			ship->getDoors().at(i)->setAngle(-1);
 			doorFree.at(i) = true;
-		} else if (doors.at(i)->resolved()) {
-			doors.at(i)->raiseDoor();
+		} else if (ship->getDoors().at(i)->resolved()) {
+			ship->getDoors().at(i)->raiseDoor();
 		}
 	}
```

Comment:
Perhaps we should think about refactoring this out somewhere else in the future, and letting `GLaDOS` solely be in charge of the generation of events.